opentelemetry-sdk/src/opentelemetry/sdk/trace/random_ids_generator.py

```diff
@@ -0,0 +1,25 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+
+from opentelemetry import trace as trace_api
+
+class RandomIdsGenerator(trace_api.IdsGenerator):
```

Comment:
Nit: Can we have this class in the same file in `ids_generator.py`.

Reply:
Definitely get what you're saying, but maybe we would want to keep it like this for these reasons:

1. Right now we have this:

```
.
├── opentelemetry-api
│   └── src
│       └── opentelemetry
│           └── trace
│               └── ids_generator.py
└── opentelemetry-sdk
    └── src
        └── opentelemetry
            └── sdk
                └── trace
                    └── random_ids_generator.py
```

I was thinking that it would be good to have `random_ids_generator.py` separate because subsequent generators could be found in the same directory level as the default "random" one. That way, all the generators would be in the same place:

```
.
├── opentelemetry-api
│   └── src
│       └── opentelemetry
│           └── trace
│               └── ids_generator.py
└── opentelemetry-sdk
    └── src
        └── opentelemetry
            └── sdk
                └── trace
                    ├── aws_xray_ids_generator.py
                    ├── bar_ids_generator.py
                    ├── foo_ids_generator.py
                    └── random_ids_generator.py
```

But as a concession java actually looks like this right now:

```
.
├── sdk
│   └── tracing
│       └── src
│           └── main
│               └── java
│                   └── io
│                       └── opentelemetry
│                           └── sdk
│                               └── trace
│                                   ├── IdsGenerator.java
│                                   └── RandomIdsGenerator.java
└── sdk_extensions
    └── aws_v1_support
        └── src
            └── main
                └── java
                    └── io
                        └── opentelemetry
                            └── sdk
                                └── extensions
                                    └── trace
                                        └── aws
                                            └── AwsXRayIdsGenerator.java
```

2. This actually follows what we already have with `span.py` because while the `span.py` has the interface in the api, the actual implementation is in the `opentelemetry.span.trace` package as below:

```
.
├── opentelemetry-api
│   └── src
│       └── opentelemetry
│           └── trace
│               └── span.py
└── opentelemetry-sdk
    └── src
        └── opentelemetry
            └── sdk
                └── trace
                    └── __init__.py
```

I like the separation of "interface" in API and "implementation" in SDK that this models!

Let me know what you think?

Reply:
It was my understanding that we would not have vendor specific generators exist in any of our packages. Ideally if users wanted to use `aws_xray_ids_generator`, the would create their own and extend from `ids_generator`. Notice how in java, the `AwsXRayIdsGenerator` is in the aws specific extension. This also forces users to take a dependency on the sdk package if they want to use the `random_ids_generator`.

Also, the design pattern of `span.py` is a special case. I believe it was done because it was just getting too large. If you look at the rest of the codebase, pretty much all classes that "relate" to each other in terms of functionality are in the same file.

Reply:
Sounds good, I think Java did it well too, and I see what you mean about `span.py`, so I'll move `RandomIdsGenerator` into the `ids_generator.py` file like you mentioned.